docker/agent/Dockerfile.agent

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG BASE_IMAGE='adoptopenjdk/openjdk8:alpine'
```

Comment:
Would this image be runnable? If not, a minimal base docker image could be used here.

Reply:
> Would this image be runnable? If not, a minimal base docker image could be used here.

I suggest `apache/skywalking-java-agent` should be runnable and be used by users like 

```dockerfile
FROM apache/skywalking-java-agent

COPY application.jar /app.jar

CMD [ "java", "-jar", "/app.jar" ]
```

instead of 


```dockerfile
FROM apache/skywalking-java-agent as agent

COPY application.jar /app.jar
COPY --from=agent /skywalking/agent /path/to/their/agent

CMD [ "java", "-javaagent:/path/to/their/agent", "-jar", "/app.jar" ]
```

the former one is more convenient and users still can use the latter one if they insist.

What do you think?

Reply:
What is the difference between this agent image and base image? @kezhenxu94

Reply:
> What is the difference between this agent image and base image? @kezhenxu94

agent image ==
  base image
  + SkyWalking Java agent (`.jars`, etc.)
  + JVM options (`-javaagent`, etc)
  + Other options / env vars that ease the use of the agent in containerization scenarios

Reply:
> Would this image be runnable? If not, a minimal base docker image could be used here.

> > What is the difference between this agent image and base image? @kezhenxu94
> 
> ```
> agent image ==
>   base image
>   + SkyWalking Java agent (`.jars`, etc.)
>   + JVM options (`-javaagent`, etc)
>   + Other options / env vars that ease the use of the agent in containerization scenarios
> ```

Will this cause the base image to lose its meaning? I think the agent image should only contains the agent files, and small enough to help users update the agent independently.
What do you think? @kezhenxu94

Reply:
> Will this cause the base image to lose its meaning?

Please note, agent image doesn't only host the agent files, and is NOT only for Kubernetes scenarios, can you take a look at https://github.com/apache/skywalking/pull/6686#issuecomment-813799007 and see whether or not it makes sense to you?